core/src/main/java/feign/ClientInvocationContext.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2012-2022 The Feign Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package feign;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ClientInvocationContext {
```

Comment:
This is something I have been thinking about for other interceptors.... having a `InvocationContext` that gives you access to all possible moving parts, instead of injecting each as parameters.  Really nice.